integration/src/main/java/scripts/testscripts/ReferencedBigQueryResourceLifecycle.java

```diff
@@ -0,0 +1,354 @@
+    referencedBqDatasetAttributes = ParameterUtils.getBigQueryDatasetReference(parameters);
+    referencedBqTableAttributes = ParameterUtils.getBigQueryDataTableReference(parameters);
+    bqTableFromAlternateDatasetAttributes =
+        ParameterUtils.getBigQueryDataTableFromAlternateDatasetReference(parameters);
+  }
+
+  @Override
+  public void doSetup(List<TestUserSpecification> testUsers, WorkspaceApi workspaceApi)
+      throws Exception {
+    super.doSetup(testUsers, workspaceApi);
+    assertThat(
+        "There must be at least three test users defined for this test.",
+        testUsers != null && testUsers.size() > 2);
+    this.noAccessUser = testUsers.get(1);
+    this.partialAccessUser = testUsers.get(2);
+  }
+
+  @Override
+  protected void doUserJourney(TestUserSpecification testUser, WorkspaceApi workspaceApi)
+      throws Exception {
+    ReferencedGcpResourceApi referencedGcpResourceApi =
+        ClientTestUtils.getReferencedGpcResourceClient(testUser, server);
+    // Grant secondary users READER permission in the workspace.
+    workspaceApi.grantRole(
+        new GrantRoleRequestBody().memberEmail(partialAccessUser.userEmail),
+        getWorkspaceId(),
+        IamRole.READER);
+    workspaceApi.grantRole(
+        new GrantRoleRequestBody().memberEmail(noAccessUser.userEmail),
+        getWorkspaceId(),
+        IamRole.READER);
+
+    // Create the references
+    GcpBigQueryDatasetResource referencedDataset =
+        BqDatasetUtils.makeBigQueryDatasetReference(
+            referencedBqDatasetAttributes,
+            referencedGcpResourceApi,
+            getWorkspaceId(),
+            MultiResourcesUtils.makeName(),
+            CloningInstructionsEnum.REFERENCE);
+    UUID bqDatasetResourceId = referencedDataset.getMetadata().getResourceId();
+    GcpBigQueryDataTableResource referencedDataTable =
+        BqDatasetUtils.makeBigQueryDataTableReference(
+            referencedBqTableAttributes,
+            referencedGcpResourceApi,
+            getWorkspaceId(),
+            MultiResourcesUtils.makeName(),
+            CloningInstructionsEnum.REFERENCE);
+    UUID bqDataTableResourceId = referencedDataTable.getMetadata().getResourceId();
+
+    // Get the references
+    GcpBigQueryDatasetResource fetchedDataset =
+        referencedGcpResourceApi.getBigQueryDatasetReference(getWorkspaceId(), bqDatasetResourceId);
+    assertEquals(referencedDataset, fetchedDataset);
+    GcpBigQueryDataTableResource fetchedDataTable =
+        referencedGcpResourceApi.getBigQueryDataTableReference(
+            getWorkspaceId(), bqDataTableResourceId);
+    assertEquals(referencedDataTable, fetchedDataTable);
+
+    // Create a second workspace to clone references into, owned by the same user
+    destinationWorkspaceId = UUID.randomUUID();
+    createWorkspace(destinationWorkspaceId, getSpendProfileId(), workspaceApi);
+    // Clone references
+    CloneReferencedGcpBigQueryDatasetResourceResult datasetCloneResult =
```

Comment:
similar to update, can we extract a cloneReferenceAndVerify

Reply:
Done - I've pulled out separate methods for get/enumerate, validate, clone, and update for each of the referenced resource lifecycle tests.